Tests and a bug fix

talvarez0507 opened the pull request:
This fixes an issue we have in master where tasks are not being sent correctly to the servlet and the button has the wrong function. It also adds tests to for some java methods.

talvarez0507 committed:
change from JSON to object

talvarez0507 committed:
fix onclick for scheduling

talvarez0507 committed:
make result list scrollable

talvarez0507 committed:
create CollectTasksFromJsonArrayTest.java

talvarez0507 committed:
create CollectEventsFromJsonArrayTest.java

talvarez0507 committed:
include algorithm dropdown

talvarez0507 commented on portfolio/src/main/webapp/task.js:
```diff
@@ -231,8 +231,7 @@ function collectAllTasks() {
         taskPriority);
-    const taskJson = JSON.stringify(task);
-    allTaskJson.push(taskJson);
+    allTaskJson.push(task);
```
This is where the bug was. collectAllTasks was returning an array of JSON objects and when information is sent to the servlet it is already converted to JSON so it was unnecessary.

talvarez0507 committed:
remove scrollable

talvarez0507 commented on portfolio/src/main/webapp/index.html:
```diff
@@ -74,17 +74,11 @@
-                Scheduling algorithm preferences
-              </a>
-              <div class="dropdown-menu" aria-labelledby="navbar-dropdown"></div>
+            <li class="nav-item nav-link">
```
This is the dropdown where a user can choose their preferred type of scheduling (the algorithm).

jamesnam-google commented on portfolio/src/main/webapp/index.html:
```diff
@@ -74,17 +74,11 @@
-                Scheduling algorithm preferences
-              </a>
-              <div class="dropdown-menu" aria-labelledby="navbar-dropdown"></div>
+            <li class="nav-item nav-link">
```
Consider putting this dropdown next or above the "Start scheduling!" button

talvarez0507 replied:
I moved it to the left of the button.

jamesnam-google commented on portfolio/src/test/java/com/google/sps/CollectEventsFromJsonArrayTest.java:
```diff
@@ -0,0 +1,56 @@
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CollectEventsFromJsonArrayTest {
```
I think this file should be named ServletHelperTest. The test file needs to be associated with the Java file.

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/test/java/com/google/sps/CollectTasksFromJsonArrayTest.java:
```diff
@@ -0,0 +1,53 @@
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CollectTasksFromJsonArrayTest {
```
Tests in this file can go into the ServletHelperTest

talvarez0507 replied:
Done.

jamesnam-google commented on portfolio/src/main/webapp/index.html:
```diff
@@ -74,17 +74,11 @@
+            <li class="nav-item nav-link">
+              Scheduling Type:
+              <select id="algorithm-type" name="algorithm-type">
+                <option value="SHORTEST_TASK_FIRST">Shortest tasks first</option>
```
This is okay, but I think the default option should be highest priority tasks first. Consider adding TODO comment.

talvarez0507 replied:
Once other algorithms are implemented they'll be added and we'll change the default option.

jamesnam-google reviewed (approved):
Approving the change with comments

talvarez0507 committed:
add ServletHelperTest.java

talvarez0507 committed:
delete file

talvarez0507 committed:
delete file

talvarez0507 committed:
scheduling row changes

Changed the scheduling button to a success button so it is green. I also moved the scheduling algorithm selection to the left of the "start scheduling" button.
